lib/x11utils.pm

```diff
@@ -112,6 +120,8 @@ sub ensure_unlocked_desktop {
                     assert_screen 'generic-desktop';
                 }
                 else {
+                    diag("Next loop ($counter), Generic desktop didn't match");
+                    record_soft_failure('bsc#1168979') if is_aarch64;
```

Comment:
@SergioAtSUSE this is the closes to the workaround that you wanted, that I find myself to agree to.

Reply:
This is wrong, the `esc` key is sent before, causing the display to redraw. bsc#1168979 causes this branch to trigger, but it is a fake. When bsc#1168979 is reproduced, the screen is locked, we should handle this before the the `esc` key. And `esc` triggers a different behavior from `shift`.

`shift` just causes the screen to redraw.

`esc` causes the lock image to roll up and show the unlock dialog.

The problem with `esc` is that the animation takes some time, in this overloaded environment and aarch64 it is too much time wasted for a workaround to detect a bug. `shift` should be used.

You can remove this part so it can be handled better in #10064

Reply:
I also think that this is most likely not working properly as the "esc" key was pressed just in before so only the *next* loop cycle should try to detect a bug condition

Reply:
See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10217#issuecomment-625882432 the `esc` does work, but it only causes the screen to be redrawn, hence the screen lock being detected in the next step (where `desktop-runner` is actually expected). One thing that comes to my mind, is pressing shift right at the beginning of the loop (or even out of it), but in doing so... we _miss/hide_ the bug, so I'd be against that

Reply:
> the `esc` does work, but it only causes the screen to be redrawn

That is wrong.

- Any key will turn on the display.
- `shift`, `ctrl`, `alt` (maybe others, I haven't checked the whole keyboard): cause the screen to redraw, showing the full screen image
- `esc` or alphanumeric key: cause the screen to redraw **and to roll up the full screen image**, showing the unlock dialog.

@ggardet, please, test is with GNOME and GDM, this is the behavior since a long time.


> hence the screen lock being detected in the next step (where `desktop-runner` is actually expected).

To avoid misleading, let me rephrase it, because the test doesn't detect the screen lock (that is why it fails), a reviewer detects it.

"hence the screen lock being shown"


>  One thing that comes to my mind, is pressing shift right at the beginning of the loop (or even out of it), but in doing so... we _miss/hide_ the bug, so I'd be against that

I agree, any key sent at the beginning of the loop hides the bug.
But, the proposal in #10064 was to check the display before pressing the key. And the content of #10064 is a the result of changing my initial proposal of including the handling in the `generic-desktop` branch.

If the test is already inside the `generic-desktop` branch, then we can send `shift`, because we already detected a `generic-desktop`. After sending that key, if we detect the screen lock ("full image" or "unlock dialog"), the bug has being detected and a soft-fail can be recorded.
And, my